examples/bridge-app/linux/include/Clusters.h

```diff
@@ -0,0 +1,408 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <functional>
+
+static constexpr uint16_t ZCL_DESCRIPTOR_CLUSTER_REVISION           = 1;
+static constexpr uint16_t ZCL_BRIDGED_DEVICE_BASIC_CLUSTER_REVISION = 1;
+static constexpr uint16_t ZCL_FIXED_LABEL_CLUSTER_REVISION          = 1;
+static constexpr uint16_t ZCL_ON_OFF_CLUSTER_REVISION               = 4;
+static constexpr uint16_t ZCL_SWITCH_CLUSTER_REVISION               = 1;
+static constexpr uint16_t ZCL_LEVEL_CONTROL_CLUSTER_REVISION        = 1;
+
+// This is the interface to cluster implementations, providing access to manipulate attributes.
+class ClusterImpl
+{
+public:
+    virtual ~ClusterImpl() = default;
+
+    virtual void SetEndpointId(chip::EndpointId id)                                                                          = 0;
+    virtual chip::ClusterId GetClusterId()                                                                                   = 0;
+    virtual EmberAfStatus Read(const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer, uint16_t maxReadLength) = 0;
+    virtual EmberAfStatus Write(const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer)                        = 0;
+    virtual void WriteFromBridge(chip::AttributeId attributeId, const uint8_t * buffer)                                      = 0;
+    virtual chip::Span<const EmberAfAttributeMetadata> GetAllAttributes()                                                    = 0;
+};
+
+// This provides a wrapper for an arbitrary-sized little-endian value that is Bytes bytes long
+// and represented logically by type T. Code can read/write it as native, or access the little-
+// endian byte array. Byte arrays also eliminate padding bytes.
+template <typename T, size_t Bytes>
```

Comment:
If Bytes is allowed to be `< sizeof(T)` and we're relying on the memcpy being ok because the "high" bytes will be 0, then we should at least be static_asserting the relevant size constraint (`Bytes <= sizeof(T)`).